Build failure: yazi

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix build github:NixOS/nixpkgs/nixos-23.11#yazi`

### Build log

```
       > error[E0599]: no method named `as_encoded_bytes` found for reference `&OsStr` in the current scope
       >   --> yazi-shared/src/fs/path.rs:51:24
       >    |
       > 51 |     let b = p.as_os_str().as_encoded_bytes();
       >    |                           ^^^^^^^^^^^^^^^^ method not found in `&OsStr`
       >
       > error[E0599]: no method named `as_encoded_bytes` found for reference `&OsStr` in the current scope
       >    --> yazi-shared/src/fs/url.rs:110:51
       >     |
       > 110 |         let path = percent_encode(self.path.as_os_str().as_encoded_bytes(), ENCODE_SET);
       >     |                                                         ^^^^^^^^^^^^^^^^ method not found in `&OsStr`
       >
       > error[E0599]: no method named `as_encoded_bytes` found for reference `&OsStr` in the current scope
       >    --> yazi-shared/src/fs/url.rs:149:40
       >     |
       > 149 |         self.file_name().map_or(false, |s| s.as_encoded_bytes().starts_with(&[b'.']))
       >     |                                              ^^^^^^^^^^^^^^^^ method not found in `&OsStr`
       >
       > For more information about this error, try `rustc --explain E0599`.
       > error: could not compile `yazi-shared` (lib) due to 3 previous errors
```

### Additional context

`as_encoded_bytes` was added in rust `1.74.0` [see here](https://doc.rust-lang.org/std/ffi/struct.OsStr.html#method.as_encoded_bytes), but nixpkgs has `1.73.0`

### Notify maintainers

@matthiasbeyer @xyenon

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.72, NixOS, 23.11 (Tapir), 23.11.20240113.428544a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/file-managers/yazi/default.nix
+++ b/pkgs/applications/file-managers/yazi/default.nix
@@ -31,18 +31,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "yazi";
-  version = "0.2.2";
+  version = "0.1.5";
 
   src = fetchFromGitHub {
     owner = "sxyazi";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-XF5zCFXiViFsRPqI6p1Z7093NSWrGmcoyWcGEagIoEA=";
+    hash = "sha256-FhKrq4N32uJRHGc0qRl+CIVNRW597jACcTFEgj8hiSE=";
   };
 
-  cargoHash = "sha256-9fXHpq5lXG9Gup1dZPlXiNilbP79fJ3Jp3+ZD7mAzP4=";
-
-  env.YAZI_GEN_COMPLETIONS = true;
+  cargoHash = "sha256-YUymZhDp1Pjm+W6m8Vmh2AgMCdaNt6TQQpiJwSg/gPw=";
 
   nativeBuildInputs = [ makeWrapper installShellFiles ];
   buildInputs = lib.optionals stdenv.isDarwin [ Foundation ];
@@ -64,9 +62,9 @@ rustPlatform.buildRustPackage rec {
       wrapProgram $out/bin/yazi \
          --prefix PATH : "${makeBinPath runtimePaths}"
       installShellCompletion --cmd yazi \
-        --bash ./yazi-config/completions/yazi.bash \
-        --fish ./yazi-config/completions/yazi.fish \
-        --zsh  ./yazi-config/completions/_yazi
+        --bash ./config/completions/yazi.bash \
+        --fish ./config/completions/yazi.fish \
+        --zsh  ./config/completions/_yazi
     '';
 
   passthru.updateScript = nix-update-script { };

